### ISSUE TITLE
JS React coloring issues

### ISSUE THREAD

**Issue by akamud** [akamud/vscode-theme-onedark#36]
As reported by @iddan in #35:

![image](https://cloud.githubusercontent.com/assets/954102/22767581/58a0d434-ee63-11e6-9c33-d388055f4569.png)

code:
```javascript
import * as styles from './mission.scss';

export default function MissionIcon({ key, title }) {
	return <i class={classnames(
		styles['mission__icon'],
		{
			[styles['mission__icon-letter']]: title.match(A_HEBREW_LETTER),
			[styles['mission__icon-first']]: !key,
		}
	)}>{title}</i>;
}
```

![image](https://cloud.githubusercontent.com/assets/954102/22767589/65b9e1d8-ee63-11e6-91db-3e1f6406bbc3.png)

Another bugs:

braces and semi-colons in yellow
decorator `@` in grey
Variables inconsistent colouring (or is it intended because they "become" keys?)
code:

```javascript
@connect(
    ({ artists }) => ({ artists }),
    dispatch => bindActionCreators({ getArtists, getArtist, addArtist, editArtist, removeArtist }, dispatch)
)
export default class AdminArtists extends Component {
    
	componentWillMount = () => {
		this.props.getArtists();
	}

	render = () => {
		console.log(this.props.artists);
		return <div>

        </div>;
	}
}
```

**Comment by akamud** [comment]
The yellow braces and semi-colons are fixed and will be present in release 1.2.0.
The inconsistency in variables are because sometimes they are properties and other special cases that get special coloring. Atom does this as well.

**Comment by vkbansal** [comment]
I've found one minor inconsistency after 1.2.0

<img width="260" alt="screen shot 2017-02-10 at 8 24 00 am" src="https://cloud.githubusercontent.com/assets/5930351/22812373/520a7e2e-ef6a-11e6-9bb4-a56d27be824a.png">

The comma between the arguments of `bar` must not be yellow.

```js
class Foo {
    static bar = (a, b) => {
        return null;
    }
}
```

**Comment by akamud** [comment]
These are all fixed in 1.3.0.
